File: internal/astgrep/testdata/fixtures/elixir/violation.ex
# Elixir violation fixture — demonstrates patterns matched by elixir rules.
# This file should produce >= 1 finding when scanned with the elixir rule set.

defmodule Example do
  def bad_code(config) do
    # Matches elixir-unused-var: assignment to _unused
    _unused = :some_value

    # Matches elixir-unsafe-map-access: Map.fetch! raises KeyError on missing key
    value = Map.fetch!(config, :required_key)

    # Matches elixir-todo-marker: stub raise
    raise "TODO"

    value
  end
end

File: internal/astgrep/testdata/fixtures/go/violation.go
package fixtures

// Violation Go fixture — demonstrates patterns matched by the shipped
// ruleset (.moai/astgrep-rules via sgconfig.yml). This file should produce
// >= 1 finding when scanned. Verified matchers on sg 0.40.5:
//   - go-interface-empty-not-any  (go/idioms.yml)
//   - sec-weak-hash-md5           (security/crypto.yml)
//   - sec-log-injection-unsanitized (security/web.yml)

import (
	"crypto/md5"
	"log"
)

// Dirty trips several shipped rules on purpose.
func Dirty(userInput string) {
	var payload interface{}
	payload = userInput
	_ = payload

	h := md5.New()
	_ = h

	log.Printf("user=%s", userInput)
}
